spyder/plugins/tests/test_editor.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+"""Tests for the Editor plugin."""
+
+# Standard library imports
+
+# Third party imports
+import pytest
+try:
+    from unittest.mock import Mock
+except ImportError:
+    from mock import Mock  # Python 2
+from qtpy.QtWidgets import QWidget
+
+# Local imports
+from spyder.utils.qthelpers import qapplication
+
+
+@pytest.fixture
+def setup_editor(qtbot, monkeypatch):
+    """Set up the Editor plugin."""
+    qapplication()
+    monkeypatch.setattr('spyder.dependencies', Mock())
+    from spyder.plugins.editor import Editor
+    monkeypatch.setattr('spyder.plugins.editor.add_actions', Mock())
+
+    class MainMock(QWidget):
+        def __getattr__(self, attr):
+            if attr.endswith('actions'):
+                return []
+            else:
+                return Mock()
+
+        def get_spyder_pythonpath(*args):
+            return []
+
+    editor = Editor(MainMock())
+    qtbot.addWidget(editor)
+    editor.show()
+
+    yield editor, qtbot
+    # teardown
+    editor.introspector.plugin_manager.close()
+
+
+def test_renamed_tree(setup_editor, mocker):
+    """Test editor.renamed_tree().
+
+    This tests that the file renaming functions are called correctly,
+    but does not test that all the renaming happens in File Explorer,
+    Project Explorer, and Editor widget as those aren't part of the plugin.
+    """
+    editor, qtbot = setup_editor
+    mocker.patch.object(editor, 'get_filenames')
```

Comment:
Just curious, mostly for my own tests—is there a reason to use `mocker.patch` vs. `monkeypatch` as is used elsewhere in the tests? I want to make sure my following best practices, and you'd likely know better than I.

Reply:
I use `mocker.patch` or `mocker.patch.object` (they are the pretty much the same -- `.object` is slightly easier to read IMHO) when I'm mocking a method or function.  `monkeypatch` is more for data attributes.

Reply:
Okay, cool. Thanks for the explanation!